### PR TITLE
🛠️ Fix specs and update deps

### PR DESCRIPTION
**Pull request by markets**
- [x] Dailymotion 
- [x] Wistia
- [x] Youtube
- [x] YoutubePlaylist
- [x] Vimeo 
- [x] Remove debugging commits (c540e6dc5c1f69d65bd7ec3d28199629e6790a8e, a575dcc46ed4291354143ba8cd7a94f9925bdd68)
- [ ] Make CI green (all passes ✅ locally) => maybe in new PR https://github.com/actions/starter-workflows/issues/785#issuecomment-770195817?

**EXTRAS**

- Update runtime deps:
  - `iso8601` from '0.9.1' to '0.13.0'
  - `oga` from '3.0' to '3.4'
  - `net_http_timeout_errors` from '0.3.0' to '1.0.0' 
- Update to latest VCR (from '3.0' to '6.1')
- Clean up gemspec and Gemfile
- Minor docs tweaks (basically Markdown format)

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -9,3 +9,5 @@ group :tool do
   gem 'ruby_gntp'
   gem 'guard-rspec'
 end
+
+gem 'byebug'
```
Comment (markets) [inline Gemfile:13]: ⚠️ **self note** remove at the end

**Comment by markets** [comment]
> What do you think about this setup?

Not really like it, because any contributor will need those keys to be able to run the tests... but I don't have any other proposal right now :) 

Since those keys are just for testing purposes, probably we can just share them with very low risk.

To be honest, I'm very lazy about creating those keys 😅... I'd like to finish this task and send some other small improvements to the project, but if someone can provide me a couple of keys that would help me a lot 👌🏼 maybe @joaocv3 (last maintainer that touched those files) have some valid keys?

**Comment by joaocv3** [comment]
Hey @markets! I actually still have them 🙏 

I'll email them to you and Thibaud.

I like the ideia of having project keys for future maintainers though. I will create a Google account and keys for the project as soon as I have some time.

**Comment by markets** [comment]
Only 1 spec failing (related to Vimeo API)!

```
Failures:

  1) VideoInfo::Providers::Vimeo with video 136971428 #view_count is expected to be > 80
     Failure/Error: it { is_expected.to be > 80 }
     
       expected: > 80
            got:   0
     # ./spec/lib/video_info/providers/vimeo_spec.rb:210:in `block (5 levels) in <top (required)>'

Finished in 17.98 seconds (files took 1.15 seconds to load)
409 examples, 1 failure, 2 pending
```

It seems the API doesn't return this data (you can try it here: https://developer.vimeo.com/api/reference/videos#get_video with this video id 136971428) anymore ... 

```
"stats": { "plays": null }
```

From what I'm reading on some issues, I think we should change some configuration in the used API key:
- https://github.com/vimeo/vimeo.php/issues/209#issuecomment-617482744
- https://stackoverflow.com/questions/51680544/of-plays-is-always-null-in-json-on-api-calls

**Pull request merged**